tck/src/main/java/org/reactivestreams/tck/PublisherVerification.java

```diff
@@ -1108,6 +1111,86 @@ public void activePublisherTest(long elements, boolean completionSignalRequired,
+   * expects a Publisher that calls onComplete.
+   * @param body the test body to execute receiving the unique publisher created via the {@link #createPublisher(long)}
+   * and the current length under test.
+   * @param elements the vararg array of elements to test, an empty array will throw an AssertionError
+   * and a single-element array delegates to {@link #activePublisherTest(long, boolean, PublisherTestRun)}.
+   * @throws Throwable allows throwing arbitrary exceptions from the test body
+   */
+  public void activePublisherMultiTest(boolean completionSignalRequired, final PublisherSubTestRun<T> body, final long... elements) throws Throwable {
+      if (elements.length == 0) {
+          throw new AssertionError("This test requires at least one length value to be specified.");
+      }
+      if (elements.length == 1) {
+          activePublisherTest(elements[0], completionSignalRequired, new PublisherTestRun<T>() {
+              @Override
+            public void run(Publisher<T> pub) throws Throwable {
+                body.run(pub, elements[0]);
+            }
+          });
+          return;
+      }
+      long max = maxElementsFromPublisher();
+      if (completionSignalRequired && max == Long.MAX_VALUE) {
+          throw new SkipException("Unable to run this test, as it requires an onComplete signal, " +
+                                    "which this Publisher is unable to provide (as signalled by returning Long.MAX_VALUE from `maxElementsFromPublisher()`)");
+        }
+      int skipped = 0;
+      List<Long> currentLength = new ArrayList<Long>();
+      List<Throwable> failures = new ArrayList<Throwable>();
+      for (long elementCount : elements) {
+          if (elementCount > max) {
+              skipped++;
+          } else {
+              env.clearAsyncErrors();
+              try {
+                  Publisher<T> pub = createPublisher(elementCount);
+                  body.run(pub, elementCount);
+                  env.verifyNoAsyncErrorsNoDelay();
+              } catch (SkipException ex) {
+                  if (ex.isSkip()) {
+                      skipped++;
+                  } else {
+                      currentLength.add(elementCount);
+                      failures.add(ex);
+                  }
+              } catch (Throwable ex) {
+                  currentLength.add(elementCount);
+                  failures.add(ex);
+              }
+          }
+      }
+      
+      if (skipped == elements.length) {
+          throw new SkipException(String.format("Unable to run this test, as all of the required elements nr: %s is higher than supported by given producer: %d", Arrays.toString(elements), maxElementsFromPublisher()));      
+      }
+      if (!failures.isEmpty()) {
+          StringBuilder sb = new StringBuilder();
+          
+          sb.append("Some of the length-specific sub-tests failed:\n");
+          
+          for (int i = 0; i < currentLength.size(); i++) {
```

Comment:
Isn't there similar print logic added elsewhere? (perhaps time to coalesce into a reusable piece)